### PR TITLE
Feature/roe 2161 enfore node 16 use

### PR DESCRIPTION
**Pull request by JoshFen95**
### JIRA link

[ROE-2161](https://companieshouse.atlassian.net/browse/ROE-2161)

### Change description

- enforce use of node16 or above and npm8 or above
- remove silent tags to provide errors in tilt window(shows error if you're not using correct node version)
- use node16 images for dockerfile

This has been locally tested on 2 other dev machines as well as my own. Pipeline changes are also ready to go. This will not be merged until pipeline changes are in place

### Work checklist

- [ ] Tests added where applicable
- [ ] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

[ROE-2161]: https://companieshouse.atlassian.net/browse/ROE-2161?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by markpit — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -16,6 +16,10 @@
     "coverage": "jest --coverage --forceExit --passWithNoTests",
     "prepare": "husky install"
   },
+  "engines": {
```
Comment (mouhajer-ch) [inline package.json:19]: We need to update `README`, `package-lock.json` and `@types/node` as described on separate chat

File: Dockerfile [hunk 2]
```diff
@@ -1,5 +1,6 @@
-FROM 169942020521.dkr.ecr.eu-west-1.amazonaws.com/base/node:14-alpine-builder
-FROM 169942020521.dkr.ecr.eu-west-1.amazonaws.com/base/node:14-alpine-runtime
+FROM 169942020521.dkr.ecr.eu-west-2.amazonaws.com/base/node-16:alpine-builder
```
[the remark this answers is not in the document]
Comment (JoshFen95) [reply Dockerfile:1]: the node16 images are only available on eu-west-2. We're moving away from eu-west-1: https://companieshouse.slack.com/archives/C2RTSL3JA/p1682315372758679?thread_ts=1682085768.145239&cid=C2RTSL3JA

[the remark this answers is not in the document]
Comment (mouhajer-ch) [reply Dockerfile:1]: Nice, probably we should add a new story related to the migration of our service from `eu-west-1` to `eu-west-2` where configured (pipeline, local development ...)

File: package.json (hunk 1 above)
Comment (JoshFen95) [reply package.json:19]: updated read me with new `node`and `npm`versions, 
generate `package-lock.json`,
update `types/node`

File: README.md [hunk 3]
```diff
@@ -42,8 +42,8 @@ Environment variables used to configure this service in docker are located in th
 
 ### Requirements
 
-1. node v12.22.12 (Concourse pipeline builds using Node 10 and live runs on Node 12)
-2. npm 6.14.16
+1. node v16 (Eegines block in package.json is used to enforce this)(Concourse pipeline builds using Node 16 and live runs on Node 16)
```
Comment (mwestacott) [inline README.md:45]: Minor spelling mistake 'engines'

File: start.sh [hunk 4]
```diff
@@ -25,5 +25,5 @@ else
     source "${APP_DIR}/global_env"
     source "${APP_DIR}/app_env"
 
-    exec node ${APP_DIR}/bin/www.js -- $PORT
+    NODE_ENV=production node_16 ${APP_DIR}/bin/www.js -- $PORT
```
Comment (ch-tstrickland) [inline start.sh:28]: According to @sroberts2 , this should read:
`    NODE_ENV=production node_16 ${APP_DIR}/dist/bin/www.js -- $PORT`

See:
https://github.com/companieshouse/admin.orders.web.ch.gov.uk/blob/main/start.sh#L33

Comment (mouhajer-ch) [reply start.sh:28]: Hi Tom, the build target is different in the two project as you can see [here](https://github.com/companieshouse/admin.orders.web.ch.gov.uk/blob/6869f7a20a3ad67ff9e8681c00bb93463a72bba6/Makefile#L55) and [here](https://github.com/companieshouse/overseas-entities-web/blob/94f7286e6a29d4502922ee7d1061a92570c0c6cc/Makefile#L38). In one we copy the `dist` directory in other the entire content.

Comment (mouhajer-ch) [reply start.sh:28]: Out of curiosity what is the difference between `node_16` and `node` command?

Comment (JoshFen95) [reply start.sh:28]: I asked Mark Ransome about this as I was curious too. He said: 
**_Yeah definitely stick with the node_16 wrapper in start.sh—when we deploy the application to the Mesos cluster there are no containers involved and the start script is the canonical way the service will be started so it needs to invoke the wrapper at that point to ensure we don’t use the default runtime (10)_**

Comment (ch-tstrickland) [reply start.sh:28]: I defer to Mou's knowledge of this app, so will resolve this.

**Review by ch-tstrickland — APPROVED** [review]

**Review by mwestacott — APPROVED** [review]

**Pull request merged**